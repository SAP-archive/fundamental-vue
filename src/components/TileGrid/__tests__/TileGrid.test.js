import { mount, createLocalVue } from "@vue/test-utils";
import FundamentalVue from "./../../../index";

const createGrid = template => {
  const localVue = createLocalVue();
  localVue.use(FundamentalVue);
  return mount(
    {
      template
    },
    { localVue }
  );
};
describe("TileGrid", () => {
  it("renders correctly with col=2", () => {
    const wrapper = createGrid(`
      <fd-tile-grid :col="2">
        <fd-tile><fd-tile-content><fd-tile-title>Tile 1</fd-tile-title></fd-tile-content></fd-tile>
        <fd-tile><fd-tile-content><fd-tile-title>Tile 2</fd-tile-title></fd-tile-content></fd-tile>
        <fd-tile><fd-tile-content><fd-tile-title>Tile 3</fd-tile-title></fd-tile-content></fd-tile>
        <fd-tile><fd-tile-content><fd-tile-title>Tile 4</fd-tile-title></fd-tile-content></fd-tile>
        <fd-tile><fd-tile-content><fd-tile-title>Tile 5</fd-tile-title></fd-tile-content></fd-tile>
      </fd-tile-grid>
    `);
    expect(wrapper.element).toMatchSnapshot();
  });

  it("renders correctly with col=6", () => {
    const wrapper = createGrid(`
      <fd-tile-grid :col="2">
        <fd-tile><fd-tile-content><fd-tile-title>Tile 1</fd-tile-title></fd-tile-content></fd-tile>
        <fd-tile><fd-tile-content><fd-tile-title>Tile 2</fd-tile-title></fd-tile-content></fd-tile>
        <fd-tile><fd-tile-content><fd-tile-title>Tile 3</fd-tile-title></fd-tile-content></fd-tile>
        <fd-tile><fd-tile-content><fd-tile-title>Tile 4</fd-tile-title></fd-tile-content></fd-tile>
        <fd-tile><fd-tile-content><fd-tile-title>Tile 5</fd-tile-title></fd-tile-content></fd-tile>
      </fd-tile-grid>
    `);
    expect(wrapper.element).toMatchSnapshot();
  });
});
