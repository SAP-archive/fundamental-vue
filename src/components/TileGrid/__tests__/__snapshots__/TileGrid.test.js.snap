// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TileGrid renders correctly with col=2 1`] = `
<div
  class="fd-tile-grid fd-tile-grid--2col"
>
  <div
    aria-disabled="false"
    class="fd-tile"
  >
    <div
      class="fd-tile__content"
    >
      <h3
        class="fd-tile__title"
      >
        Tile 1
      </h3>
    </div>
  </div>
   
  <div
    aria-disabled="false"
    class="fd-tile"
  >
    <div
      class="fd-tile__content"
    >
      <h3
        class="fd-tile__title"
      >
        Tile 2
      </h3>
    </div>
  </div>
   
  <div
    aria-disabled="false"
    class="fd-tile"
  >
    <div
      class="fd-tile__content"
    >
      <h3
        class="fd-tile__title"
      >
        Tile 3
      </h3>
    </div>
  </div>
   
  <div
    aria-disabled="false"
    class="fd-tile"
  >
    <div
      class="fd-tile__content"
    >
      <h3
        class="fd-tile__title"
      >
        Tile 4
      </h3>
    </div>
  </div>
   
  <div
    aria-disabled="false"
    class="fd-tile"
  >
    <div
      class="fd-tile__content"
    >
      <h3
        class="fd-tile__title"
      >
        Tile 5
      </h3>
    </div>
  </div>
</div>
`;

exports[`TileGrid renders correctly with col=6 1`] = `
<div
  class="fd-tile-grid fd-tile-grid--2col"
>
  <div
    aria-disabled="false"
    class="fd-tile"
  >
    <div
      class="fd-tile__content"
    >
      <h3
        class="fd-tile__title"
      >
        Tile 1
      </h3>
    </div>
  </div>
   
  <div
    aria-disabled="false"
    class="fd-tile"
  >
    <div
      class="fd-tile__content"
    >
      <h3
        class="fd-tile__title"
      >
        Tile 2
      </h3>
    </div>
  </div>
   
  <div
    aria-disabled="false"
    class="fd-tile"
  >
    <div
      class="fd-tile__content"
    >
      <h3
        class="fd-tile__title"
      >
        Tile 3
      </h3>
    </div>
  </div>
   
  <div
    aria-disabled="false"
    class="fd-tile"
  >
    <div
      class="fd-tile__content"
    >
      <h3
        class="fd-tile__title"
      >
        Tile 4
      </h3>
    </div>
  </div>
   
  <div
    aria-disabled="false"
    class="fd-tile"
  >
    <div
      class="fd-tile__content"
    >
      <h3
        class="fd-tile__title"
      >
        Tile 5
      </h3>
    </div>
  </div>
</div>
`;
